scheduler-interface-spec.md

```diff
@@ -891,21 +891,23 @@ message EventRecord {
      REQUEST_ALLOC      = 101;  // Request allocated
      REQUEST_TIMEOUT    = 102;  // Request cancelled due to timeout
 
-     APP_ALLOC          = 200;  // Allocation changed
-     APP_REQUEST        = 201;  // Request changed
-     APP_REJECT         = 202;  // Application rejected on create
-     APP_NEW            = 203;  // Application added with state new
-     APP_ACCEPTED       = 204;  // State change to accepted 
-     APP_STARTING       = 205;  // State change to starting
-     APP_RUNNING        = 206;  // State change to running
-     APP_COMPLETING     = 207;  // State change to completing
-     APP_COMPLETED      = 208;  // State change to completed
-     APP_FAILING        = 209;  // State change to failing
-     APP_FAILED         = 210;  // State change to failed
-     APP_RESUMING       = 211;  // State change to resuming
-     APP_EXPIRED        = 212;  // State change to expired
-     APP_CANNOTRUN      = 213;  // Application cannot run (maxApplications hit)
-     APP_CANRUN         = 214;  // Application is allowed to run (after maxApplications limit hit)
+     APP_ALLOC            = 200;  // Allocation changed
+     APP_REQUEST          = 201;  // Request changed
+     APP_REJECT           = 202;  // Application rejected on create
+     APP_NEW              = 203;  // Application added with state new
+     APP_ACCEPTED         = 204;  // State change to accepted
+     APP_STARTING         = 205;  // State change to starting
+     APP_RUNNING          = 206;  // State change to running
+     APP_COMPLETING       = 207;  // State change to completing
+     APP_COMPLETED        = 208;  // State change to completed
+     APP_FAILING          = 209;  // State change to failing
+     APP_FAILED           = 210;  // State change to failed
+     APP_RESUMING         = 211;  // State change to resuming
+     APP_EXPIRED          = 212;  // State change to expired
+     APP_CANNOTRUN_QUEUE  = 213;  // Application cannot run in the queue (maxApplications hit)
+     APP_RUNNABLE_QUEUE   = 214;  // Application is allowed to run (after maxApplications limit hit)
+     APP_CANNOTRUN_QUOTA  = 215;  // Application cannot run due to user/group quota (maxApplications hit)
+     APP_RUNNABLE_QUOTA   = 216;  // Application is allowed to run based on user/group quota (after limit hit)
```

Comment:
More detailed constants

Reply:
Are there some copy/paste issues in these descriptions? 3 of the 4 new ones refer to maxApplications...

Reply:
No, it's all related to maxapps. I'll fix the last one to keep consistency.